Start export task when the user confirms the transfer

suzannehamilton opened the pull request:
When the user fills out the confirmation form, call the API to trigger the consignment export and show the user the "Transfer complete" page. There's an open question about whether we should show the user a transfer progress page instead, but we'll implement that later if necessary.

There are a few refactoring commits before the main change, so it's probably easiest to review commit-by-commit.

This is a draft because it shouldn't be merged until https://github.com/nationalarchives/tdr-prototype-sangria/pull/25 is deployed.

suzannehamilton commented on app/model/ReviewData.scala:
```diff
@@ -0,0 +1,3 @@
+package model
+
+case class ReviewData(confirmRecordTransfer: String, confirmOpen: String, confirmTnaOwnership: String)
```
I've copied the format of the form data from the service agreement page, but I think these fields should be booleans rather than Strings. Is there a reason that they're not booleans? @TomJKing - do you know?

TomJKing replied:
They should be booleans.

suzannehamilton replied:
As discussed in person, these should be booleans but we're going to leave them as they are in the prototype and fix them if we have time or if makes sense when implementing something else.

TomJKing commented on app/controllers/ReviewController.scala:
```diff
@@ -17,19 +22,60 @@ class ReviewController @Inject()(
+
+    val formValidationResult: Form[ReviewData] = form.bindFromRequest
 
-      val appSyncClient = client.graphqlClient
```
Probably should not use the val name "appSyncClient" as no longer using app sync.

suzannehamilton replied:
Good point. I've renamed them all. :+1:

TomJKing commented on app/controllers/ReviewController.scala:
```diff
@@ -47,8 +49,17 @@ class ReviewController @Inject()(
+
     val successFunction: ReviewData => Future[Result] = { formData: ReviewData =>
-      Future.apply(Redirect(routes.TransferConfirmationController.index()))
+      val appSyncClient = client.graphqlClient
```
Probably should not use the val name "appSyncClient" as not using app sync anymore

suzannehamilton replied:
Fixed.

TomJKing reviewed (approved):
Couple of minor comments.

suzannehamilton committed:
Show user the confirmation page when they confirm a transfer

Link the confirmation form to the transfer confirmation page. This
doesn't update any data or start any export tasks yet.

suzannehamilton committed:
Update generated GraphQL schema

Add user fields and consignment transfer confirmation.

suzannehamilton committed:
Delete unused auth headers parameter

This header was used by the OAuth Cognito code which has been removed.

suzannehamilton committed:
Confirm transfer in API when user clicks 'Transfer' button

Send a confirmation request to the API, which updates the progress
of the transfer and triggers an export.

suzannehamilton committed:
Remove unnecessary futures from ServiceAgreementsController

suzannehamilton committed:
Rename appSyncClient variables to graphQlClient

We are no longer prototyping appSync, so remove all references to it.

suzannehamilton committed:
Replace `Future.apply` with `Future.successful` in controllers

In each case, we already have the data to return but it needs to be
wrapped in a Future. `Future.apply` and `Future.successful` both
wrap data in a future, but `Future.successful` is a better option in
these cases because it wraps the data in a Future synchronously on
the current thread. `Future.apply` starts an asyncronous process,
which is unnecessary unless you are performing another async
operation.
